kanbananza-backend/src/main/controllers/card.js

```diff
@@ -77,6 +77,23 @@ const update = async (req, res, next) => {
       );
     }
 
+    if (req.body.order !== undefined) {
+      const cards = cardService.findCardsWithLargerOrder(req.params.id);
+      let myCard;
+      let lastIncreasedOrder = req.body.order;
+      let i;
+      for (i = 0; i < cards.length; i += 1) {
+        myCard = cards[i];
+
+        if (myCard.order > lastIncreasedOrder) {
+          break;
+        }
+
+        cardService.updateCardById(req.params.id, { order: myCard.order + 1 });
+        lastIncreasedOrder += 1;
```

Comment:
lastIncreasedOrder  += 1  only makes sense if you set lastIncreasedOrder to the myCard.order at the beginning of each loop. 

otherwise you may have


| i | myCard.order | lastIncreasedOrder |
| - | --------------| -------------------|
| 3 | 6 | 5 | 
| 4 | 12 | 6 | 
| 5 | 13 |  7 | 

its inconsistent because the order can grow by more than one

Reply:
then mycard.order would be larger and it would break